### PR TITLE
🎉 New Connector: Fleetio source

### PR DESCRIPTION
**Pull request by jmmizerany**
## What
- Adding in a new source connector to pull data from the [Fleetio API](https://developer.fleetio.com/)
- For context on what fleetio is, the website is linked [here](https://fleetio.com/)
- The only UI change is adding the Fleetio Source connector , along with our Fleetio icon, to the source collection for connectors

## How
- This connector was built using the airbyte low-code CDK
- Currently this support full table replication only, but in future version we will introduce incremental syncing

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

- No breaking changes to Airbyte or this connector

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: .github/workflows/terminate-zombie-build-instances.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # Also a failsafe against a dev writing a workflow that does not terminate build instances.
 # Though the average Airbyte build runtime as of this commit is ~20 mins, connector builds
 # can take up to 3 hours. Set this to 3 hours to include these longer runs.
-name: Terminate Zombie Build Instances
+- name: Terminate Zombie Build Instances
```
Comment (natikgadzhi) [inline .github/workflows/terminate-zombie-build-instances.yml:5]:
```suggestion
name: Terminate Zombie Build Instances
```

Seems to be an unrelated change?

File: airbyte-integrations/connectors/source-fleetio/.dockerignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+*
+!Dockerfile
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-fleetio/.dockerignore:2]: Note to self: we should remove `!Dockerfile`, `!setup.py` and friends once we complete the migration to Poetry for all sources.

File: airbyte-integrations/connectors/source-fleetio/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
+FROM python:3.9.11-alpine3.15 as base
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-fleetio/Dockerfile:1]: Since this is a new source, let's get it to use Airbyte-provided base image. That'd ensure that it's up to date, and will get docker image security updates as well, without any changes to the connector itself.

To get this done: 
- [ ] Remove the Dockerfile
- [ ] Set `connectorBuildOptions` in the `metadata.yaml`. [Example in `source-mailchimp`](https://github.com/airbytehq/airbyte/blob/d41cd621e77651af29e5d74cc0e0611514fb4c3a/airbyte-integrations/connectors/source-mailchimp/metadata.yaml#L8):

   ```
     connectorBuildOptions:
       baseImage: docker.io/airbyte/python-connector-base:1.1.0@sha256:bd98f6505c6764b1b5f99d3aedc23dfc9e9af631a62533f60eb32b1d3dbab20c
   ```

The current version is `1.2.0` though, double-check SHA on DockerHub.

File: airbyte-integrations/connectors/source-fleetio/README.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+This is the repository for the Fleetio configuration based source connector.
+For information about how to use this connector within Airbyte, see [the documentation](https://docs.airbyte.com/integrations/sources/fleetio).
+
+## Fleetio Overview
+View the Fleetio website [here](https://fleetio.com). For for information the Fleetio API, check out out [Fleetio API Portal](https://developer.fleetio.com).
+
+## Local development
+
+#### Building via Gradle
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-fleetio/README.md:11]: @alafanechere, whoa whoa whoa, I think we don't use Gradle for connectors anymore, right? Should we replace this with `airbyte-ci` instructions? Can you copy over the snippet in here in a suggestion, please?

File: airbyte-integrations/connectors/source-fleetio/README.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+to generate the necessary credentials. Then create a file `secrets/config.json` conforming to the `source_fleetio/spec.yaml` file.
+Note that any directory named `secrets` is gitignored across the entire Airbyte repo, so there is no danger of accidentally checking in sensitive information.
+See `integration_tests/sample_config.json` for a sample config file.
+
+**If you are an Airbyte core member**, copy the credentials in Lastpass under the secret name `source fleetio test creds`
+and place them into `secrets/config.json`.
+
+### Locally running the connector docker image
+
+#### Build
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-fleetio/README.md:30]: The whole build block is now replaced with `airbyte-ci connectors --name source-fleetio build`.

File: airbyte-integrations/connectors/source-fleetio/README.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+#### Acceptance Tests
+Customize `acceptance-test-config.yml` file to configure tests. See [Connector Acceptance Tests](https://docs.airbyte.com/connector-development/testing-connectors/connector-acceptance-tests-reference) for more information.
+If your connector requires to create or destroy resources for use during acceptance tests create fixtures for it and place them inside integration_tests/acceptance.py.
+
+To run your integration tests with Docker, run:
+```
+./acceptance-test-docker.sh
+```
+
+### Using gradle to run tests
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-fleetio/README.md:62]: Seems like the whole file needs to be updated. One way to do this is to regenerate a new connector with new generator from `master`, or check other new-ish up to date sources. Mailchimp is a decent example.

File: airbyte-integrations/connectors/source-fleetio/acceptance-test-docker.sh [hunk 7]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env sh
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-fleetio/acceptance-test-docker.sh:1]: Please delete this file. We're now using `airbyte-ci` for tests.

File: airbyte-integrations/connectors/source-fleetio/build.gradle [hunk 8]
```diff
@@ -0,0 +1,9 @@
+plugins {
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-fleetio/build.gradle:1]: Yeah, let's delete Gradle build from here. We're walking away from using Gradle for anything Python.

File: airbyte-integrations/connectors/source-fleetio/requirements.txt [hunk 9]
```diff
@@ -0,0 +1,2 @@
+-e ../../bases/connector-acceptance-test
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-fleetio/requirements.txt:1]: Let's drop requirements.txt (delete the file) in favor or Poetry and `pyproject.toml`. I'm happy to help! Basically, we're going away from `setup,py` and `requirements.txt` and using Poetry for each connector. 

You can pretty much copy and paste this: https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mailchimp/pyproject.toml — but make sure to NOT pin the CDK version. The new one is 0.65.something I believe.

**Review by natikgadzhi — CHANGES_REQUESTED** [review]
Hey @jmmizerany, huge thank you for contributing the connector!

Overall, this looks good. We're migrating some pieces to new infra (removing gradle, Poetry, base docker image) and I've requested changes to align with the new approach, to make sure the connector will be easy to maintain in the future. 

I'm happy to help you through them as well. Once we get those cleaned up, let's also setup a test account for testing on our CI, and let's get this merged! 👏

File: airbyte-integrations/connectors/source-fleetio/README.md (hunk 4 above)
Comment (alafanechere) [reply airbyte-integrations/connectors/source-fleetio/README.md:11]: [I suggest following a structure like we have on `source-stripe` 
](https://github.com/airbytehq/airbyte/blob/main/airbyte-integrations/connectors/source-stripe/README.md#L37)

File: .github/workflows/terminate-zombie-build-instances.yml (hunk 1 above)
Comment (jmmizerany) [reply .github/workflows/terminate-zombie-build-instances.yml]: @natikgadzhi Sorry about this, I was able to undo this accidental commit so it should not be apart of this PR anymore

**Comment by jmmizerany** [comment]
Hey all, is there an update on the changes I have pushed? We are wanting to get this live as we have a beta test with some of our end users coming up

**Review by natikgadzhi — APPROVED** [review]
This looks good to me, high level. @marcosmarxm, can you get this over the finish line? What are our requirements for new connectors? Should we obtain a sandbox account and store it internally for tests?

@jmmizerany, would you be able to provide us with a sandbox account that Airbyte could use to run acceptance tests for this connector? I'm not sure how we usually share those credentials ;)

**Comment by jmmizerany** [comment]
> This looks good to me, high level. @marcosmarxm, can you get this over the finish line? What are our requirements for new connectors? Should we obtain a sandbox account and store it internally for tests?
> 
> @jmmizerany, would you be able to provide us with a sandbox account that Airbyte could use to run acceptance tests for this connector? I'm not sure how we usually share those credentials ;)

Thanks! We have a sandbox account ready to go. Let me know how we can coordinate the next steps

**Comment by marcosmarxm** [comment]
I ran tests with provided sandbox credentials and some tests are failing. Later this week I'll return with a review and update of changes.

**Comment by jmmizerany** [comment]
> I ran tests with provided sandbox credentials and some tests are failing. Later this week I'll return with a review and update of changes.

Thank you @marcosmarxm Please let me know what I can do to help

File: airbyte-integrations/connectors/source-fleetio/source_fleetio/schemas/contacts.json [hunk 10]
```diff
@@ -0,0 +1,240 @@
+{
+    "$schema": "http://json-schema.org/schema#",
+    "properties": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-fleetio/source_fleetio/schemas/contacts.json:3]: all objects type must have `additionalProperties: true`

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Remove the spec.yaml file as you're setting the spec inside the manifest.yaml

**Review by natikgadzhi — COMMENTED** [review]
Let's get this over the finish line, but then — we should also inline all schemas from json files into the manifest. We have a new tool that can assist with that, `airbyte-ci connectors --name source-fleetio migrate_to_inline_schemas` in https://github.com/airbytehq/airbyte/pull/37649.

**Comment by jmmizerany** [comment]
> Let's get this over the finish line, but then — we should also inline all schemas from json files into the manifest. We have a new tool that can assist with that, `airbyte-ci connectors --name source-fleetio migrate_to_inline_schemas` in #37649.

Thanks!

**Review by natikgadzhi — APPROVED** [review]
I think for a low-code connector with a bunch of integration tests, this is good enough. For failing cats, this might stop us from doing new changes to this connector (which I would advice — like inlining schemas). 

We will reach out if we need to fix them up.
